rfcs/2021-11-22-design-tokens.md

```diff
@@ -0,0 +1,108 @@
+
+3. When the first dataset is completed and ready for use, a tool stored in the NPM package would need to be created to convert these tokens into new formats, initially SASS. There are several existing tools which can do this such as [Style Dictionary](https://github.com/amzn/style-dictionary) (format is [easily utilized by the DSM](https://support.invisionapp.com/hc/en-us/articles/360049483032)), [Theo](https://github.com/salesforce-ux/theo), [JSON -> CSS](https://github.com/rlapoele/json-to-scss) and others.
+
+   I believe the best option would be to write our own [Webpack Loader](https://webpack.js.org/contribute/writing-a-loader/) to convert these tokens to top level SASS files.
+
+   1. In the case of the DS-Core, we would convert these tokens into variables which would be utilized by files in our /src/styles/settings/variables scss files and font.scss file initially. Since we would not be changing these files aside from referencing the token variable names in the tokens NPM package instead of providing hard coded values, the compiled output of these files should be compared to detect any breaking changes. Token variables would reside in scss files, checked into the token package. One advantage to using a custom webpack loader in our case is that we could do things like generate the list of @font-face includes based on what tokens are available, slimming overall DS package size.
+   2. In the case of Sketch and the DSM, the tool chosen/created would be extended to output libraries for the DSM (if needed) and Sketch and these libraries would also be versioned and live in the token package. These would be made available as release download libraries with which to work from. The [Sketch file format](https://developer.sketch.com/file-format/) is very easy to work with as it is already made up of JSON files.
+
+4. A Jenkins task for creating these releases would need to be created and tested for CI of changes to the tokens package.
+5. Future work to structure similarities between component level variables in sketch and the codebase would be ongoing from this point. These variables could be added and utilized on a component by component basis, and would make use of the highest level tokens for spacing, color, etc.. i.e.
+
+   ```
+   button-small-active-highlight-inverse-color (sketch)
+   $button-small-active-highlight-inverse-color (scss)
+
+   tokens.git/core/components/buttons.json
+   {
+   	"button": {
+   		"small": {
+   			"active": {
+   				"highlight": {
+   					"color": "ds.core.colors.lightblue.300"
+   				}
+   			}
+   		}
+   	}
+   }
+   ```
+
+6. Documentation would need to be generated from the token build task which could be another option in our existing site docs filed under "Design Tokens." Component level variables could be included on the doc pages for each component.
```

Comment:
```suggestion
6. Documentation would need to be generated from the token build task. This could be another option in our existing site docs, filed under "Design Tokens." Component level variables could be included on the doc pages for each component.
```

Reply:
What would the documentation cover?